### DISCUSSION TITLE
[Feature] Small thumbnail size

### DISCUSSION THREAD

**Discussion by neonoviy** [immich-app/immich#4455] [Feature Request]
### The feature

Hi! I've got large photo gallery with thousands of photos. It can be hundreds of photos each day. So I have to scroll a lot looking for a photo I want. I can change page zoom to 25% and it's fine size for thumbnails of me, but, of course, other parts of interface became unusable. I think it would be great option, to make thumbnails smaller. Much smaller. Maybe to let user input thumbnail size. More photos on one page, less space on disk.

### Platform

- [ ] Server
- [X] Web
- [ ] Mobile

**Comment by MikNZAU** [comment]
Yes, this would be great. Even a slider at the top that the user can move for small medium large extra-large  thumbs.

I propose we don;t need to make actually 4 different size thumbs, stick with current, but just change the number of columns in the grid.

**Comment by lukakiro** [comment]
I started developing a solution: https://github.com/immich-app/immich/pull/6238

Any suggestion is welcome

**Comment by DaveTSG** [comment]
This would be a fantastic addition to Immich. Thank you @lukakiro for giving it a try.
[This is a video](https://www.dropbox.com/scl/fi/kskkbowvxsfg7gwi3klle/PhotosThumbnailZoom.mp4?rlkey=gr7aqymvr1r6qllgo9ddwxqxz&dl=0) of what I can do with Apple Photos on MacOS. It would be great if something similar could occur in Immich.

**Reply by blublub** [reply]
A slider in top of the page would be awesome to adjust the displayed thumbnail size!

Additional option:
show a larger thumbnail on when I stay over it with the mouse when thumbnails are show "small"

**Reply by pgfpcsxwqr-ctrl** [reply]
Many of us use Immich with very large photo/video libraries (100k+ assets). Browsing such large collections on mobile apps and in the web browser becomes extremely slow and inefficient when the grid is limited to 6 items per row, especially on tablets.

It would be very useful to add flexible grid density options for mobile apps and the web UI, allowing users to choose smaller thumbnail sizes and more items per row. Apple Photos on iPad can display around 27 or even 67 items per row depending on the selected thumbnail size, and on iPhone it supports approximately 15 or 30 items per row with thumbnail sizes around 78×78 and 38×38 respectively.

**Comment by basicuser01** [comment]
I am still searching for this basic feature. Thank you!

**Comment by MikNZAU** [comment]
Just a bump for this feature. Desperately would love to see it!

**Comment by tkafka** [comment]
+1, I foud this issue when researching if I could have thumbnail size slider in main Immich view. For searching for specific photos, the default view is far too large.

**Comment by ok1ok1** [comment]
I used AI to create a tamper monkey script that does this pretty well. I'm using immich version 3.2.0.
This version of the script only works on the main timeline but it could be adapted for other pages. It's a good starting point for anyone that wants to tinker. 

The slider is pinned to the top bar next to the upload button. It scales the thumbnails from 35%-150%
It does not scale the special first row of thumbnails used for the auto curated filters (1 year ago, 2 years ago, etc...)

Feel free to download it from the link below. 

https://gist.github.com/ok1ok1/3ca2f79df8b7237c6776105a4d05e62c
